File: episodes/002-clean-code-naming/src/python/overdue.py
"""Episode 002 — intention-revealing names (Python)."""

from datetime import date


def days_overdue(due_date: date, returned_date: date) -> int:
    """Return how many days late a book was returned (0 if on time)."""
    late_days = (returned_date - due_date).days
    return max(late_days, 0)


def main() -> None:
    due = date(2026, 7, 1)
    returned = date(2026, 7, 12)
    print(f"Days overdue: {days_overdue(due, returned)}")


if __name__ == "__main__":
    main()

File: episodes/002-clean-code-naming/src/javascript/overdue.js
// Episode 002 — intention-revealing names (JavaScript)

const MS_PER_DAY = 1000 * 60 * 60 * 24;

/**
 * Return how many days late a book was returned (0 if on time).
 * @param {Date} dueDate
 * @param {Date} returnedDate
 * @returns {number}
 */
function daysOverdue(dueDate, returnedDate) {
  const lateDays = Math.round((returnedDate - dueDate) / MS_PER_DAY);
  return Math.max(lateDays, 0);
}

function main() {
  const due = new Date("2026-07-01");
  const returned = new Date("2026-07-12");
  console.log(`Days overdue: ${daysOverdue(due, returned)}`);
}

if (require.main === module) {
  main();
}

module.exports = { daysOverdue };
